mso/utils.go

```diff
@@ -398,3 +400,32 @@ func GetPolicyByName(cont *container.Container, policyName string, templateEleme
 
 	return nil, fmt.Errorf("Policy name %s not found", policyName)
 }
+
+func testAccVerifyKeyValue(resourceAttrsMap *map[string]string, resourceAttrRootkey, stateKey, stateValue string) {
+	stateKeySplit := strings.Split(stateKey, ".")
+	for inputKey, inputValue := range *resourceAttrsMap {
+		if strings.Contains(stateKey, resourceAttrRootkey) && stateKeySplit[len(stateKeySplit)-1] == inputKey && (stateValue == inputValue || (inputValue == "reference" && stateValue != "")) {
+			delete(*resourceAttrsMap, inputKey)
+			break
+		}
+	}
+}
+
+func customTestCheckResourceAttr(resourceName, resourceAttrRootkey string, resourceAttrsMap map[string]string) resource.TestCheckFunc {
+	return func(is *terraform.State) error {
+		rootModule, err := is.RootModule().Resources[resourceName]
+		if !err {
+			return fmt.Errorf("%v", err)
+		}
+		if rootModule.Primary.ID == "" {
+			return fmt.Errorf("No ID is set for the template")
+		}
+		for stateKey, stateValue := range rootModule.Primary.Attributes {
+			testAccVerifyKeyValue(&resourceAttrsMap, resourceAttrRootkey, stateKey, stateValue)
+		}
+		if len(resourceAttrsMap) > 0 {
```

Comment:
What is the purpose of this condition?

Reply:
The `resourceAttrsMap` is our input value, which is `map[string]string`, when the `resourceAttrsMap` entry is matching with the state file content I will remove the matched entries from the `resourceAttrsMap`. 

The `resourceAttrsMap` should be empty when it matches all the input values with the state file content otherwise we have something which is not matching with the state file content and we fail the test with not matched input value.